internal_ws/ykbh/src/lib.rs

```diff
@@ -195,8 +195,8 @@ impl SIRInterpreter {
     /// the block where the guard failed, we immediately skip to the terminator and interpret it to
     /// see which block we need to start interpretation in.
     pub unsafe fn interpret(&mut self, tio: *mut u8) {
-        // Set the pointer to the trace inputs.
-        self.set_trace_inputs(tio);
+        // Set the pointer to the trace ctx.
+        self.set_trace_ctx(tio);
```

Comment:
Right of the bat, I thought we were going for "interpreter context", not "trace context".

Rationale is that this is something the interpreter author has to define themselves and here we are in the black-hole interpreter, which doesn't deal with traces at all.

So is "interpreter context" a name we'd rather use?

Maybe you've renamed `tio` later, but this was a shorthand for "trace IO". In newer code I've pre-empted your renaming and started using `icx` as a shorthand for `interpreter context`.

Reply:
Good point. Yes, `set_interp_ctx` would make more sense. Fixed in https://github.com/softdevteam/yk/pull/222/commits/c8814fb14de802599e5d607124aed9384064aa93.

Reply:
`ctx` is fine, but we should grep for `icx` and rename them to `ctx`.

(Unless you'd rather use `icx`)

Reply:
There are no `icx`s that I can see?

Reply:
I found a handful:
```
internal_ws/ykshim/src/prod_api.rs
112:unsafe extern "C" fn __ykshim_si_interpret(si: *mut ykbh::SIRInterpreter, icx: *mut u8) {
114:    si.interpret(icx);

internal_ws/ykshim/src/test_api.rs
114:unsafe extern "C" fn __ykshimtest_interpret_body(body_name: *mut c_char, icx: *mut u8) {
117:    si.set_trace_inputs(icx);

ykshim_client/src/prod_api.rs
38:    fn __ykshim_si_interpret(interp: *mut RawSIRInterpreter, icx: *mut u8);
137:    pub unsafe fn execute(&self, icx: &mut I) -> *mut RawSIRInterpreter {
139:        f(icx)

ykshim_client/src/test_api.rs
49:    fn __ykshimtest_interpret_body(body_name: *mut c_char, icx: *mut u8);
141:pub fn interpret_body<I>(body_name: &str, icx: &mut I) {
143:    unsafe { __ykshimtest_interpret_body(body_cstr.into_raw(), icx as *mut _ as *mut u8) };
```

Reply:
Looks like you got them in a later commit, sorry again.